code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,276 +40,184 @@ internal sealed class AEPPushTemplate {
     internal val body: String
 
     // Required, Version of the payload assigned by the authoring UI.
-    internal var payloadVersion: Int?
-        private set
+    internal val payloadVersion: Int?
 
     // begin optional values
     // Optional, sound to play when the notification is shown
-    internal var sound: String?
-        private set
+    internal val sound: String?
 
     // Optional, number to show on the badge of the app
-    internal var badgeCount: Int = 0
-        private set
+    internal val badgeCount: Int
 
     // Optional, priority of the notification
-    internal var notificationPriority: Int?
-        private set
-
-    // Optional, importance of the notification. Only used on Android N and above.
-    private var notificationImportance: Int?
```

Comment:
Is `importance` no longer important?!

Reply:
Lol XD We need map the priority to importance for API level > N and set it to the notification channel in `createNotificationChannelIfRequired`

Reply:
Lol...
I am keeping the only keeping the properties in AEPPushTemplate that we only read from notification payload (here we read as priorityString)

This value translates to both `NotificationPriority` for API O and below and `NotificationImportance` For API N and above

I kept two methods that would translate the priorityString to appropriate importance and priority value
```
   @RequiresApi(api = Build.VERSION_CODES.N)
    fun getNotificationImportance(): Int {}
    
    fun getNotificationPriority(): Int {}
  ```